### PR TITLE
[ios] send result when mail has send

### PR DESCRIPTION
**Pull request by robindijkhof**
fixes https://github.com/taljacobson/flutter_mailer/issues/26

### CODE REVIEW

**Comment by robindijkhof** [comment]
@taljacobson any idea why the labeler has failed?

**Comment by taljacobson** [comment]
firstly **thank you** for the PR.

as with the labeler, not sure yet.
just gives an error of 
```
##[error]HttpError: Resource not accessible by integration

```
not a big deal if it doesn't work.

as for the PR, i wont have the ability to test it for on a ios device for atleast a week to QA it out.
so might just merge and push as a dev release. if that is not a problem.

**Comment by taljacobson** [comment]
ok published as `0.5.1-dev.1`.

with that im thinking we might make this a major release, with breaking changes.
the ability to know if the email was sent, canceled or saved in ios could be very beneficial for some.
the only draw back is the it is only possible in ios. for android we might just want to send a string of `unknown | Android`

by changing the dart `send` signature from a Future that returns `<void>` to `<String>` or provide a enum with `Sent | Canceled | Draft | Android`

**Comment by robindijkhof** [comment]
That 's fine

Op di 9 jun. 2020 22:32 schreef Tal Jacobson <notifications@github.com>:

File: ios/Classes/FlutterMailerPlugin.m [hunk 1]
```diff
@@ -172,6 +174,7 @@ - (void)mailComposeController:(MFMailComposeViewController *)controller didFinis
     while (ctrl.presentedViewController && ctrl != controller) {
         ctrl = ctrl.presentedViewController;
     }
+    flutterResult(nil);
```
Comment (taljacobson) [inline ios/Classes/FlutterMailerPlugin.m:177]: we should probably add a protection measure here, to make sure it is not null.

**Comment by taljacobson** [comment]
merged this as 0.5.1-dev.2 and published as 0.5.1-dev.2.
will change to a non dev version soon.

if you can use the dev version and report back that it works as intended on a physical device, that would be much apriciated.

**Comment by robindijkhof** [comment]
I tested my branch on a physical device. Are there any other changes?

**Comment by taljacobson** [comment]
none other then pubspec and changelog.

**Comment by taljacobson** [comment]
published as 0.5.1 
no need for a dev version

**Pull request merged**